app/models/loader/orchestrate.rb

```diff
@@ -255,13 +255,63 @@ def insert_new
       @new_roles = ::Role.all
 
       in_primary_schema do
-        TABLES.each do |table|
-          columns = (TABLE_EQUIVALENCE_COLUMNS[table] + [ :policy_id ]).join(", ")
-          db.execute "INSERT INTO #{table} ( #{columns} ) SELECT #{columns} FROM #{schema_name}.#{table}"
-        end
+        disable_policy_log_trigger
+        TABLES.each { |table| insert_table_records(table) }
+        enable_policy_log_trigger
       end
     end
 
+    def insert_table_records(table)
+      columns = (TABLE_EQUIVALENCE_COLUMNS[table] + [ :policy_id ]).join(", ")          
+      db.run "INSERT INTO #{table} ( #{columns} ) SELECT #{columns} FROM #{schema_name}.#{table}"
+      
+      # For large policies, the policy logging triggers occupy the majority
+      # of the policy load time. To make this more efficient on the initial
+      # load, we disable the triggers and update the policy log in bulk.
+      insert_policy_log_records(table)
+    end
+
+    def disable_policy_log_trigger
+      # To disable the triggers during the bulk load we use a local
+      # configuration setting that the trigger function is aware of. 
+      # When we set this variable to `true`, then the trigger will
+      # observe the setting value and skip its own policy log.
+      db.run 'SET LOCAL conjur.skip_insert_policy_log_trigger = true'
+    end
+
+    def enable_policy_log_trigger
+      db.run 'SET LOCAL conjur.skip_insert_policy_log_trigger = false'
+    end
+
+    def insert_policy_log_records(table)
+      db.run <<-POLICY_LOG
+          INSERT INTO policy_log(
+            policy_id, 
+            version,
+            operation, 
+            kind, 
+            subject)
+          SELECT
+            r.policy_id,
+            r.version,
+            r.operation,
+            r.kind,
+            r.subject
+          FROM
+            #{schema_name}.#{table},
+            policy_log_#{table}_record(
+              /* We need to convert the temp schema table type
+                 to the publish schema type for the same table.
+                 We have to use string serialization to accomplish
+                 this. */
+              #{table}::text::#{table},
```

Comment:
@apotterri, do you have any ideas on a better way to accomplish this?

The two types have the same column definitions, but I couldn't find a way to directly cast the temporary schema type to the public schema without marshaling through `text`.

This doesn't appear to impact the performance in a meaningful way, it just feels unpolished to me.